common/src/main/java/org/esbtools/message/admin/common/dao/MetadataDAOImpl.java

```diff
@@ -290,9 +303,55 @@ private MetadataField searchField(MetadataField tree, MetadataField field) {
     }
 
     @Override
-    public void sync(String entity, String system, String key, String... values) {
-        // TODO Auto-generated method stub
+    public MetadataResponse sync(String entity, String system, String key, String... values) {
 
+        MetadataResponse result = new MetadataResponse();
+        // create JMS Payload
+        StringBuilder message = new StringBuilder("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+        message.append("<SyncRequest><EntityName>");
+        message.append(entity);
+        message.append("</EntityName><KeyName>");
+        message.append(key);
+        message.append("</KeyName>");
+        for(String value: values) {
+            if(value!=null && value.length()>0) {
+                message.append("<KeyValue>");
+                message.append(value);
+                message.append("</KeyValue>");
+            }
+        }
+        message.append("</SyncRequest>");
+        log.log(Level.INFO, "Initiating sync request:"+message.toString());
+
+        boolean foundActiveHost = false;
+        for(String restEndPoint: resyncRestEndpoints) {
+
+            try {
+                URL url = new URL(restEndPoint);
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+                conn.setDoOutput(true);
+                conn.setRequestMethod("POST");
+                conn.setRequestProperty("Content-Type", "application/xml");
+                OutputStream os = conn.getOutputStream();
+                os.write(message.toString().getBytes());
+                os.flush();
+                if (conn.getResponseCode() == HttpURLConnection.HTTP_OK) {
+                    foundActiveHost = true;
+                    break;
+                } else {
+                    log.warning("unable to send resync message to:"+restEndPoint);
+                }
+                conn.disconnect();
+            } catch (MalformedURLException e) {
+                result.setErrorMessage(e.getMessage());
+            } catch (IOException e) {
+                result.setErrorMessage(e.getMessage());
+            }
+        }
+        if(!foundActiveHost && result.getStatus()==MetadataResponse.Status.Success) {
```

Comment:
I understand that the implementation of the status will allow this to work, however the code is still confusing.  If I were to read this code aloud, it would sound something like this

"If I didn't find an active host and the status is successful..."

The reason I was confused initially was that not finding an active host seems like something that would make the status of a response invalid.  Also, in the case where the server returns something other than a 200 (like a 500 for instance), the foundActiveHost variable will remain false (even though we actually did find an active host, which returned an error).

It seems like we could eliminate the foundActiveHost variable altogether if the statuses were explicitly set based on the response from the http call (instead of assuming nothing changed the status since the initialization)